examples/cpp_api/nullable_attribute.cc

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file   nullable_attribute.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2020 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * When run, this program will create a simple 2D dense array with one fixed
+ * nullable attribute and one var-sized nullable attribute, write some data
+ * to it, and read the data back on both attributes.
+ */
+
+#include <iostream>
+#include <tiledb/tiledb>
+
+using namespace tiledb;
+
+// Name of array.
+std::string array_name("nullable_atrributes_array");
+
+void create_array() {
+  // Create a TileDB context
+  Context ctx;
+
+  // The array will be 2x2 with dimensions "rows" and "cols", with domain [1,2]
+  Domain domain(ctx);
+  domain.add_dimension(Dimension::create<int>(ctx, "rows", {{1, 2}}, 2))
+      .add_dimension(Dimension::create<int>(ctx, "cols", {{1, 2}}, 2));
+
+  // The array will be dense
+  ArraySchema schema(ctx, TILEDB_DENSE);
+  schema.set_domain(domain).set_order({{TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR}});
+
+  // Create two attributes "a1" and "a2", the first fixed and the second
+  // variable-sized.
+  Attribute a1 = Attribute::create<int>(ctx, "a1");
+  Attribute a2 = Attribute::create<std::vector<int>>(ctx, "a2");
+
+  // Set both attributes as nullable
+  a1.set_nullable(true);
+  a2.set_nullable(true);
+
+  schema.add_attribute(a1);
+  schema.add_attribute(a2);
+
+  // Create the (empty) array on disk.
+  Array::create(array_name, schema);
+}
+
+template <typename T>
```

Comment:
this is totally good code, but I'd want to avoid advanced C++ in the examples if we can help it. I think we should remove this routine and do something a little more straight-forward. I'll comment below.

Reply:
ok sure, I am removing this